app/views/courses/index.html.erb

```diff
@@ -0,0 +1,20 @@
+<% @title = t('activerecord.models.course.other') %>
+<h1><%= @title %></h1>
+<table class="course-table">
+  <tbody>
+    <% @courses.in_groups_of(4).each do |group| %>
+      <tr class="row">
+        <% group.each do |course| %>
+        <td class="course-item col-md-3">
+          <% if course %>
+          <%= image_tag course.subject.image %> <br>
+          <div class="course-name">
+            <%= t("subjects.#{course.subject.name}") %>
+          </div>
```

Comment:
Try to render one view for teachers and students but in lines 10-13 render different partials